teuthology/provision/__init__.py

```diff
@@ -20,6 +16,7 @@ def _logfile(ctx, shortname):
 
 
 def reimage(ctx, machine_name):
+    from teuthology.provision import fog
```

Comment:
Why is this required, that is not a good practice, looks like you're trying to resolve circular imports here, where are you facing this issue.

Reply:
no. i am not trying to fix cyclic dependency issue. please see the commit message. i am trying to minimize the set of dependencies required for running `vstart_runner.py`, which does not need to talk to a remote service for provision, etc.

Reply:
I don't see how it could talk to remote service for provision until you call corresponding methods, what's wrong with those dependencies in the vstart? Still I don't think to place imports in the function is a good idea.

Reply:
no, it does not use the openstack API until the methods are called, but the crux is that i want to avoid the import. because it's a waste of time and resource to install unused modules, just because they are imported in the beginning of the `__init__.py`.

> Still I don't think to place imports in the function is a good idea.

could you be more specific why it is not a good idea? yes, it's unusual. but why is it bad? if you insist, i will have to partially invert my changes in `setup.py` and move some "orchestra" back into "install_requires".

Reply:
seems some unit tests are using them. so i end up moving them to "install_requires".